code/modules/admin/player_panel/actions/punish.dm

```diff
@@ -152,3 +152,123 @@
 
 	var/mob/living/carbon/Xenomorph/X = target
 	X.generate_name()
+
+/datum/player_action/reset_human_name
+	action_tag = "reset_human_name"
+	name = "Reset Human Name"
+
+
+/datum/player_action/reset_human_name/act(var/client/user, var/mob/target, var/list/params)
+	var/mob/target_mob = target
+	if(!ismob(target_mob))
+		to_chat(user, SPAN_WARNING("[target.name] is not a mob!"))
+		return
+
+	if(isXeno(target_mob))
+		to_chat(user, SPAN_WARNING("[target.name] is a xeno!"))
+		return
+
+	if(!target_mob.client)
+		to_chat(user, SPAN_WARNING("[target.name] does not have a client!"))
+
+	if(!target_mob.ckey)
+		to_chat(user, SPAN_DANGER("Warning: Mob ckey for [target_mob.name] not found."))
+		return
+
+	if(alert(user, "Are you sure you want to reset name for [target_mob.ckey]?", "Confirmation", "Yes", "No") == "No")
+		return
+
+	var/new_name
+
+	if(alert(user, "Do you want to manually set the name for [target_mob.ckey]?", "Confirmation", "Yes", "No") == "No")
```

Comment:
should add a third option here that lets em cancel out if they got this far and regret it